internal/extproc/translator/jsonschema_helper.go

```diff
@@ -0,0 +1,408 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package translator
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"google.golang.org/genai"
+)
+
+// get the supported field names
```

Comment:
please write complete sentences for any function documentation and what is "supported fields"? could you clarify?

```suggestion
// jsonSchemaGetAllowedFields gets ... foo bar.
```

Reply:
made it a variable